### ISSUE TITLE
Store shipping: Combine two packages sections into one

### ISSUE THREAD

**Issue by kellychoffman** [Automattic/wp-calypso#16610]
Originally came up in: https://github.com/Automattic/wp-calypso/pull/16540. In WCS in core admin, there are two packages sections, breaking up custom and flat rate packages services like USPS provide for free at the post office.

Seeing the two sections in WP.com makes it clear that more work is needed refining this area. The merchant shouldn't have to go through and manage packages two different ways depending on where the package came from and what type it is. 

Here's one solution that takes the existing UI elements and rearranges them a bit to be more cohesive:

Merchant lands on Shipping page and clicks Add Package:
![00](https://user-images.githubusercontent.com/5835847/28648728-21be900c-722d-11e7-886c-4186b734ff12.png)
Merchant can add their own custom package:
![01](https://user-images.githubusercontent.com/5835847/28648729-21c0791c-722d-11e7-8e55-e51fb6dd51d8.png)
Or choose from one of the flat rate packages provided by a service
![02](https://user-images.githubusercontent.com/5835847/28648731-21c79b7a-722d-11e7-8cb9-a09135f7566a.png)
Package has been added:
![04](https://user-images.githubusercontent.com/5835847/28648730-21c13a50-722d-11e7-9395-5c52f4056306.png)

cc: @marcinbot and @jameskoster for design and implementation feedback.

**Comment by jameskoster** [comment]
Design looks really good. I'm going to list the nitpicks I noted originally in https://github.com/Automattic/wp-calypso/pull/16540 as this is partly a design issue and they're _seemingly_ small changes. If they're out of scope that's fine, lemme know and I'll create another issue.

* Let's use our [dimensions input](https://cloudup.com/coRWPD3-s34) for the package inner/outer dimensions.
* "View exterior dimensions" suggests I'm going to look at something when I click that link. I actually wondered for a second if it was going to generate a 3d version of the package for me to preview, based on the inner dimensions I supplied 🤓. Can we make it "Add exterior dimensions"? When editing a package, if exterior dimensions exist just display them.
* Weight inputs should use input-with-suffix component.

**Comment by kellychoffman** [comment]
> Let's use our dimensions input for the package inner/outer dimensions.

Agree. I will confess I screencapped our current implementation in WCS here.

> Can we make it "Add exterior dimensions"? When editing a package, if exterior dimensions exist just display them.

That makes more sense. Let's do that. 

> Weight inputs should use input-with-suffix component.

Agree.

**Comment by allendav** [comment]
Where can I find flat rate packaging?

<img width="967" alt="screen shot 2017-08-16 at 10 23 37 am" src="https://user-images.githubusercontent.com/1595739/29376334-11f1cc5c-826d-11e7-846d-492ecb54e25a.png">

**Comment by allendav** [comment]
I tried adding a custom package after adding two service packages, but it looks like it didn't actually "take"

![custom-package-fails-to-appear](https://user-images.githubusercontent.com/1595739/29376693-444afda8-826e-11e7-8d72-2c36448c78d6.gif)

**Comment by marcinbot** [comment]
> Where can I find flat rate packaging?

Flat rates were removed from the predefined package list a while ago in WCS. There's a 1:1 mapping of flat rate boxes to flat rate services, so selecting a flat rate service in WCS also "selects" the corresponding package.

> I tried adding a custom package after adding two service packages, but it looks like it didn't actually "take"

You can use the modal to either add a custom package or a set of predefined ones. Can't do both. I wonder if it wouldn't confuse people though 🤔

**Comment by allendav** [comment]
I am pretty sure this is a bug - the custom package appears if you delete a package - probably a rendering/state bug?

![custom-package-fails-to-appear-then-appears-when-you-remove-a-service-package](https://user-images.githubusercontent.com/1595739/29378734-06fbe442-8275-11e7-8276-2bf2dd06ec45.gif)

**Comment by allendav** [comment]
> Flat rates were removed from the predefined package list a while ago in WCS. There's a 1:1 mapping of flat rate boxes to flat rate services, so selecting a flat rate service in WCS also "selects" the corresponding package.

Ah - cool!

**Comment by marcinbot** [comment]
> I am pretty sure this is a bug - the freshly added package appears ONLY AFTER you delete a package - probably a rendering/state bug?

It was a problem with memoization. I pushed a fix.
